calc/src/mechanics/util.ts

```diff
@@ -356,6 +357,34 @@ export function getFinalDamage(
   return OF16(pokeRound(Math.max(1, OF32(damageAmount * finalMod) / 0x1000)));
 }
 
+/**
+ * Determines which move category Shell Side Arm should behave as. Applications
+ * that consume this library may want to import this function to show users move
+ * category information.
+ *
+ * Notes:
+ * 1. As far as I can tell, research hasn't been tone on whether it's the larger
+ * ratio (atk / def) or the difference (atk - def) which decides the move
+ * category.
+ * Ratio seems more likely to me, since the pokemon damage formula is
+ * dependent on the ratio and not the difference of attack and defense stats.
+ *
+ * 2. in game, when stats are equal, category is random.  For damage, it won't
+ * matter most of the time (items like assault vest aren't included in this
+ * decision though, so *may* take effect).
+ *
+ * {@link https://www.reddit.com/r/VGC/comments/hcfe4k/galarian_slowbro_shell_side_arm_breakdown/}
```

Comment:
https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/page-51#post-8514607

Here is a post containing the research on the move.